lua/autorun/ragdollremover.lua

```diff
@@ -11,92 +11,132 @@ local RemoveSpeed = 4000        -- Velocity ragdoll is removed at
  
-local function SetSubPhysMotionEnabled( ent, enable )
-        if not IsValid( ent ) then return end
-	
-        ent:SetVelocity( vector_origin )
+local entMeta = FindMetaTable("Entity") -- Why? Both ways work but I think the looks cleaner :P
+
+function entMeta:SetSubPhysMotion( enable )
+	if not IsValid( self ) then return end
 	
-        if ( not enable ) then
-                ent:SetColor( Color( 255, 0, 255, 255 ) )
-                if IsValid( ent:GetOwner() ) then
-                        if ( IsTTT ) then
-                                ent:GetOwner():GetWeapon( "weapon_zm_carry" ):Reset( false )
-                        end
-                end
-        else
-                ent:SetColor( Color( 255, 255, 255, 255 ) )
-        end
+	if ( not enable ) then
+		self:SetColor( Color( 255, 0, 255, 255 ) )
+		if IsValid( self:GetOwner() ) and IsTTT then
+			self:GetOwner():GetWeapon( "weapon_zm_carry" ):Reset( false )
+		end
+	else
+		self:SetColor( Color( 255, 255, 255, 255 ) )
+	end
 	
-        for i = 0, ent:GetPhysicsObjectCount() - 1 do
-                local subphys = ent:GetPhysicsObjectNum( i )
-                if IsValid( subphys ) then
-                        subphys:EnableMotion( enable )
-                        if ( not enable ) then
-                                subphys:SetVelocity( vector_origin )
-                                subphys:SetMass( subphys:GetMass() * 20 )
-                        else
-                                subphys:SetMass( subphys:GetMass() / 20 )
-                                subphys:Wake()
-                        end
-                end
+	for i = 0, self:GetPhysicsObjectCount() - 1 do
+		local subphys = self:GetPhysicsObjectNum( i )
+		if IsValid( subphys ) then
+			subphys:EnableMotion( enable )
+			if ( not enable ) then
+				subphys:SetVelocity( vector_origin )
+				subphys:SetMass( subphys:GetMass() * 20 )
+			else
+				subphys:SetMass( subphys:GetMass() / 20 )
+				subphys:Wake()
+			end
+		end
 	end
-
-        ent:SetVelocity( vector_origin )
+	self:SetVelocity( vector_origin )
 end
  
 local function KillVelocity( ent )
-	SetSubPhysMotionEnabled( ent, false )
+	if not IsValid(ent) then return end
+	
+	ent:SetSubPhysMotion(false)
 	
 	if ( EchoFreeze ) then 
 		PrintMessage( HUD_PRINTTALK, "[GS_CRASH] Body frozen for " .. FreezeTime .. " seconds!" ) 
 	end
 	
-	timer.Simple( FreezeTime, function() 
-		SetSubPhysMotionEnabled( ent, true )
-		if ( EchoFreeze ) then 
-			PrintMessage( HUD_PRINTTALK, "[GS_CRASH] Body unfrozen!" )
+	timer.Simple(FreezeTime, function()
+		if IsValid(ent) then -- Just in case
+			ent:SetSubPhysMotion(true)
+			if ( EchoFreeze ) then 
+				PrintMessage( HUD_PRINTTALK, "[GS_CRASH] Body unfrozen!" )
+			end
 		end
-	end )
+	end)
 end
 
+local bodyfound = GetConVar("ttt_announce_body_found", "1")
+
 local function IdentifyCorpse( corpse )
-        if not ( corpse ) then return end
+	if not IsValid(corpse) or CORPSE.GetFound(corpse, false) then return end
```

Comment:
Okay, I see what the parameter does now. I'll keep it as true so the function passes even during fallback since there are plenty of checks inside of the function for validity. Thanks